tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/Serverless.cs

```diff
@@ -0,0 +1,96 @@
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(Serverless));
+
+        internal static void InitIfNeeded()
+        {
+            if (IsRunningInLambda(ExtensionFullPath))
+            {
+                Debug("Sending CallTarget serverless integration definitions to native library.");
+                var serverlessDefinitions = GetServerlessDefinitions();
+                NativeMethods.InitializeProfiler(DefinitionsId, serverlessDefinitions);
+                foreach (var def in serverlessDefinitions)
+                {
+                    def.Dispose();
+                }
+
+                Debug("The profiler has been initialized with serverless definitions, count = " + serverlessDefinitions.Length);
+            }
+        }
+
+        internal static bool IsRunningInLambda(string extensionPath)
+        {
+            string pathFromEnv = EnvironmentHelpers.GetEnvironmentVariable(ExtensionEnvName);
+            string path = pathFromEnv != null ? pathFromEnv : extensionPath;
+            return File.Exists(path) && !string.IsNullOrEmpty(EnvironmentHelpers.GetEnvironmentVariable(FunctionEnvame));
+        }
+
+        internal static NativeCallTargetDefinition[] GetServerlessDefinitions()
+        {
+            try
+            {
+                LambdaHandler handler = new LambdaHandler(EnvironmentHelpers.GetEnvironmentVariable(HandlerEnvName));
+                string assymblyName = typeof(InstrumentationDefinitions).Assembly.FullName;
+                string integrationType = GetIntegrationTypeFromParamCount(handler.ParamTypeArray.Length);
+                return new NativeCallTargetDefinition[]
+                {
+                    new(handler.GetAssembly(), handler.GetFullType(), handler.GetMethodName(), handler.ParamTypeArray, 0, 0, 0, 65535, 65535, 65535, assymblyName, integrationType)
+                };
+            }
+            catch (Exception ex)
+            {
+                Serverless.Error("Impossible to get Serverless Definitions : " + ex.Message);
+                return Array.Empty<NativeCallTargetDefinition>();
+            }
+        }
+
+        internal static string GetIntegrationTypeFromParamCount(int paramCount)
+        {
+            int inputParamCount = paramCount - 1; // since the return type is in the array
+            switch (inputParamCount)
+            {
+                case 0:
+                    return typeof(AWS.LambdaNoParam).FullName;
+                case 1:
+                    return typeof(AWS.LambdaOneParam).FullName;
+                case 2:
+                    return typeof(AWS.LambdaTwoParams).FullName;
+                default:
+                    throw new ArgumentOutOfRangeException("AWS Lambda handler number of params can only be 0, 1 or 2.");
+            }
+        }
+
+        internal static void Debug(string str)
+        {
+            if (EnvironmentHelpers.GetEnvironmentVariable(LogLevelEnvName).ToLower() == "debug")
```

Comment:
You don't want to allocate a string everytime you write a lot statement:
```suggestion
            if (string.Equals(EnvironmentHelpers.GetEnvironmentVariable(LogLevelEnvName), "debug", StringComparison.OrdinalIgnoreCase)
```